malcolm/modules/pmac/parts/pmacchildpart.py

```diff
@@ -759,6 +760,141 @@ def get_some_points(self, start_index):
 
         return points, joined, velocities
 
+    def check_profile_length_exceeds_profile_points(self) -> bool:
+        # Check if we have exceeded the points number and need to write
+        # Strictly less than so we always add one more point to the time
+        # array so we can always stretch points in a subsequent add with
+        # the values already in the profiles
+        return len(self.profile["timeArray"]) > PROFILE_POINTS
+
+    def create_generator_profile_every_point(self, start_index: int) -> bool:
```

Comment:
This looks quite similar to `create_generator_profile_sparse()`. How much code do they share? Could the be re-combined to a single function or does that make it unreadable? If not, could some common code be extracted into other functions?